component/config.go

```diff
@@ -31,46 +33,131 @@ type ConfigValidator interface {
 // ValidateConfig validates a config, by doing this:
 //   - Call Validate on the config itself if the config implements ConfigValidator.
 func ValidateConfig(cfg Config) error {
-	return validate(reflect.ValueOf(cfg))
+	var err error
+	errs := validate(reflect.ValueOf(cfg))
+
+	for _, suberr := range errs {
+		if suberr.err != nil {
```

Comment:
When does this happen?

Reply:
I previously allowed `nil` errors so we didn't have to do a nil check every time we called `callValidateIfPossible` on the value. I'm okay either way, so I've updated it so we don't include `pathError` objects with nil `err` fields in the slice anymore.